pkg/cloudprovider/provider/alibaba/provider.go

```diff
@@ -0,0 +1,439 @@
+	request.ResourceType = "instance"
+	tags := []ecs.AddTagsTag{tag}
+	request.Tag = &tags
+
+	if _, err := client.AddTags(request); err != nil {
+		return fmt.Errorf("failed to create new UID tag: %v", err)
+	}
+
+	return nil
+}
+
+func (p *provider) SetMetricsForMachines(machines v1alpha1.MachineList) error {
+	return nil
+}
+
+func (p *provider) getConfig(s v1alpha1.ProviderSpec) (*Config, *providerconfigtypes.Config, error) {
+	if s.Value == nil {
+		return nil, nil, errors.New("machine.spec.providerconfig.value is nil")
+	}
+	pconfig := providerconfigtypes.Config{}
+	err := json.Unmarshal(s.Value.Raw, &pconfig)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	rawConfig := alibabatypes.RawConfig{}
+	if err = json.Unmarshal(pconfig.CloudProviderSpec.Raw, &rawConfig); err != nil {
+		return nil, nil, err
+	}
+
+	c := Config{}
+	c.AccessKeyID, err = p.configVarResolver.GetConfigVarStringValueOrEnv(rawConfig.AccessKeyID, "ALIBABA_ACCESS_KEY_ID")
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get the value of \"AccessKeyID\" field, error = %v", err)
+	}
+	c.AccessKeySecret, err = p.configVarResolver.GetConfigVarStringValueOrEnv(rawConfig.AccessKeySecret, "ALIBABA_ACCESS_KEY_SECRET")
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get the value of \"AccessKeySecret\" field, error = %v", err)
+	}
+	c.InstanceType, err = p.configVarResolver.GetConfigVarStringValue(rawConfig.InstanceType)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get the value of \"instanceType\" field, error = %v", err)
+	}
+	c.RegionID, err = p.configVarResolver.GetConfigVarStringValue(rawConfig.RegionID)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get the value of \"regionID\" field, error = %v", err)
+	}
+	c.VSwitchID, err = p.configVarResolver.GetConfigVarStringValue(rawConfig.VSwitchID)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get the value of \"vSwitchID\" field, error = %v", err)
+	}
+	c.ZoneID, err = p.configVarResolver.GetConfigVarStringValue(rawConfig.ZoneID)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get the value of \"zoneID\" field, error = %v", err)
+	}
+	c.InternetMaxBandwidthOut, err = p.configVarResolver.GetConfigVarStringValue(rawConfig.InternetMaxBandwidthOut)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get the value of \"internetMaxBandwidthOut\" field, error = %v", err)
+	}
+	c.Labels = rawConfig.Labels
+	return &c, &pconfig, err
+}
+
+func getClient(regionID, accessKeyID, accessKeySecret string) (*ecs.Client, error) {
+	client, err := ecs.NewClientWithAccessKey(regionID, accessKeyID, accessKeySecret)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get Alibaba cloud client: %v", err)
+	}
+	return client, nil
+}
+
+func getInstance(client *ecs.Client, instanceName string) (*ecs.Instance, error) {
+	describeInstanceRequest := ecs.CreateDescribeInstancesRequest()
+	describeInstanceRequest.InstanceName = instanceName
```

Comment:
Please use the UID label to identify your instance instead of the name

Reply:
addressed here: https://github.com/kubermatic/machine-controller/pull/643/files#diff-e96854fadd3b15014f29f87ac940a01bR234-R247

Reply:
I don't follow, how does that code address anything? The code here still filters the retrieved instance only based on the name and not on the tags